### PR TITLE
Stream deck plugin v3

### PR DESCRIPTION
**Pull request by fcannizzaro**
Hello! I'm back 😄

After the update of the Elgato Stream Deck application (6.5) the old plugin broke so I rewrote it with their new SDK (for node finally)  / features and replaced some old interactions with simpler ones.

Main changes:
- removed the old authentication flow (popup notification/number confirmation), now there is an "**Authorize application**" button that opens a now a Stream Deck Application deep link.  
- removed notification for selection item/loadout, now when a selection is required from the Stream Deck I enable the dnd on item/loadout/in-game loadout, and the old information is sent through the `dataTransfer` in the dnd event.

flow

- removed old SD unused code / no more needed LS keys
- #9586 

*Maybe there will be some other changes*

As always @bhollis LMYWYT but don't merge this PR since I'm waiting for the Elgato team to start working again to send them the plugin update.

### CODE REVIEW

**Comment by bhollis** [comment]
I love seeing all the stuff that's removed :-)

One of the design goals for the original plugin integration was that if someone wasn't using the streamdeck integration, we wouldn't load the code. I can't quite tell from reading this whether that's still mostly true?

**Comment by fcannizzaro** [comment]
> I love seeing all the stuff that's removed :-)
> 
> One of the design goals for the original plugin integration was that if someone wasn't using the streamdeck integration, we wouldn't load the code. I can't quite tell from reading this whether that's still mostly true?

Yeah mostly, but there are two functions `useStreamDeckLoadoutSelection`, and the code inside the draggable item that I'm thinking how to isolate in the async module.

File: src/app/stream-deck/StreamDeckSettings/StreamDeckSettings.tsx [hunk 1]
```diff
@@ -1,22 +1,39 @@
...
+  let result = '';
+  const characters = 'ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789';
+  const charactersLength = characters.length;
+  let counter = 0;
+  while (counter < length) {
+    result += characters.charAt(Math.floor(Math.random() * charactersLength));
+    counter += 1;
+  }
+  return result;
+};
```
Comment (bhollis) [inline src/app/stream-deck/StreamDeckSettings/StreamDeckSettings.tsx:33]: Does the token have to be in this format, or can it also be a UUID?

File: src/app/stream-deck/StreamDeckSettings/StreamDeckSettings.tsx [hunk 2]
```diff
@@ -1,22 +1,39 @@
...
+
+const STREAM_DECK_DEEP_LING = 'streamdeck://plugins/message/com.dim.streamdeck';
```
Comment (bhollis) [inline src/app/stream-deck/StreamDeckSettings/StreamDeckSettings.tsx:35]:
```suggestion
const STREAM_DECK_DEEP_LINK = 'streamdeck://plugins/message/com.dim.streamdeck';
```

File: src/app/stream-deck/async-module.ts [hunk 3]
```diff
@@ -208,37 +73,35 @@ function refreshStreamDeck(): ThunkResult {
...
-        body: t('StreamDeck.Authorization.Update'),
-        type: 'error',
-        duration: 200,
-        onClick: () => notificationPromise.resolve(),
-        promise: notificationPromise.promise,
-      });
-    }
-    dispatch(streamDeckUpdatePopupShowed());
+// refresh the stream deck state on refresh events
+const installRefresObserver = _.once(() => {
```
Comment (bhollis) [inline src/app/stream-deck/async-module.ts:85]:
```suggestion
const installRefreshObserver = _.once(() => {
```

**Review by bhollis — APPROVED** [review]
Looks great. Would there be a way to detect that the user's streamdeck plugin is outdated and tell them to upgrade? I remember last time we got a lot of confused people in our Discord...

**Comment by fcannizzaro** [comment]
> Looks great. Would there be a way to detect that the user's streamdeck plugin is outdated and tell them to upgrade? I remember last time we got a lot of confused people in our Discord...

I reintroduce the version check with all cases

If all is working (connected and authorization already setup)
image

if there is a older version (or any other reason that block the connection, since in my discord server someone had issue with adblocker using Brave)
image

if is connected but the authentication is missing
image

This button in the header is lazy loaded with a Suspense only if the plugin is enabled.

File: config/i18n.json [hunk 4]
```diff
@@ -1289,24 +1289,22 @@
...
   },
   "StreamDeck": {
-    "Connected": "Connected",
-    "NotConnected": "Not Connected",
+    "Tooltip": {
+      "Title": "DIM Stream Deck Plugin",
+      "Version": "Version:",
+      "Application": "Stream Deck Application",
+      "Plugin": "Plugin",
+      "Error": "Your Stream Deck plugin is no more supported. Please update to the latest version. This plugin requires at least:",
```
Comment (delphiactual) [inline config/i18n.json:1297]:
```suggestion
      "Error": "Your Stream Deck plugin is no longer supported. Please update to the latest version. This plugin requires at least:",
```

**Review by delphiactual — APPROVED** [review]

**Comment by fcannizzaro** [comment]
plugin published✅

**Pull request merged**